src/Features/Core/Portable/Navigation/NavigableItemFactory.SymbolLocationNavigableItem.cs

```diff
@@ -38,8 +41,21 @@ private class SymbolLocationNavigableItem : INavigableItem
 
             public bool IsImplicitlyDeclared => _symbol.IsImplicitlyDeclared;
 
-            public Document Document
-                => _location.IsInSource ? _solution.GetDocument(_location.SourceTree) : null;
+            public INavigableItem.NavigableDocument Document
+            {
+                get
+                {
+                    return LazyInitialization.EnsureInitialized(
+                        ref _lazyDocument,
+                        static self =>
+                        {
+                            return (self._location.IsInSource && self._solution.GetDocument(self._location.SourceTree) is { } document)
+                                ? INavigableItem.NavigableDocument.FromDocument(document)
+                                : null;
```

Comment:
this is concerning.  We can return null, but the API says it's not null.

Reply:
It's not clear which item you are referring to (there are two return values in question).

This overload of `EnsureInitialized` operates on a `StrongBox<T>` specifically because it allows the initialized value to be null.

https://github.com/dotnet/roslyn/blob/806dba8b9e1cdfe271adc02f748362cc7af98f13/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Utilities/LazyInitialization.cs#L89-L101

Reply:
> It's not clear which item you are referring to 

Api is: `public INavigableItem.NavigableDocument Document` which is non-null.  my guess is the file is not NRT annotated though.  If that's the case, you don't have to change anything.

If this is NRT annotated, something is wrong :)

Reply:
This file does not have null annotations enabled currently.